client/utilities/rfc-index-xml.ts

```diff
@@ -0,0 +1,213 @@
+  abortController: AbortController
+  redApi: ApiClient
+  delayBetweenRequestsMs: number
+}
+
+export async function renderRfcIndexDotXml(props: Props) {
+  const { push } = props
+  push('<?xml version="1.0" encoding="UTF-8"?>\n')
+  push(
+    '<rfc-index xmlns="https://www.rfc-editor.org/rfc-index" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://www.rfc-editor.org/rfc-index https://www.rfc-editor.org/rfc-index.xsd">\n'
+  )
+  // await renderBCPs(props)
+  // await renderFYIs(props)
+  await renderRFCs(props)
+  // await renderSTDs(props)
+
+  push('</rfc-index>')
+  close()
+}
+
+type XMLBuilderOptions = NonNullable<
+  ConstructorParameters<typeof XMLBuilder>[0]
+>
+
+const xmlBuilderOptions: XMLBuilderOptions = {
+  ignoreAttributes: true,
+  attributeNamePrefix: '',
+  format: true,
+  suppressBooleanAttributes: true,
+  indentBy: '  ',
+  textNodeName: '#text'
+}
+
+// const renderBCPs = async (props: Props): Promise<void> => {
+//   console.log(props.delayBetweenRequestsMs) // FIXME: remove this
+//   // FIXME: render BCPs
+// }
+
+// const renderFYIs = async (props: Props): Promise<void> => {
+//   console.log(props.delayBetweenRequestsMs) // FIXME: remove this
+//   // FIXME: render FYIs
+// }
+
+const renderRFCs = async ({
+  push,
+  abortController,
+  redApi,
+  delayBetweenRequestsMs
+}: Props): Promise<void> => {
+  const docListArg: DocListArg = {}
+  docListArg.sort = ['-number'] // sort by oldest RFC to find the end
+  docListArg.limit = 1 // we only need one result
+  const response = await redApi.red.docList(docListArg)
+  const largestRfcNumber = response.results[0].number
+  // const longestRfcNumberLength = largestRfcNumber.toString().length
+
+  const builder = new XMLBuilder({
+    ...xmlBuilderOptions,
+    arrayNodeName: 'rfc-entry'
+  })
+
+  docListArg.sort = ['number'] // sort by first RFC
+  let offset = 0
+
+  while (!abortController.signal.aborted) {
+    docListArg.offset = offset
+    docListArg.limit = 1000
+
+    const response = await redApi.red.docList(docListArg)
+
+    response.results.forEach((rfcMetadata) => {
+      const rfc = getRFCWithExtraFields(rfcMetadata)
+
+      const [month, year] = DateTime.fromISO(rfcMetadata.published)
+        .toFormat('LLLL yyyy')
+        .split(' ')
+
+      // Based on https://github.com/rfc-editor/rpcwebsite/blob/edf4896c1d97fdd79a78ee6145e3a0c5ffb11fb9/rfc-ed/bin/xmlIndex.pl
+      const rfcForXml = {
+        'rfc-entry': {
+          'doc-id': `RFC${rfc.number}`,
+          title: rfc.title,
+          ...(rfc.authors.length > 0 ?
+            {
+              author: {
+                name: rfc.authors.map(formatAuthor)
+              }
+            }
+          : {}),
+          date: { month, year },
+          ...(rfc.formats ?
+            {
+              format: {
+                'file-format': rfc.formats.map((format) => {
+                  if (format.type === 'TXT') {
+                    return 'ASCII'
```

Comment:
Just pointing since I think this is the conversion point we talked about on slack. I think it's fine to do this for now and amend later when (I hope) we get the ok to leave it 'TXT' everywhere.